src/walletConnect/saga.ts

```diff
@@ -196,9 +201,21 @@ function* handleInitialisePairing({ uri, origin }: InitialisePairing) {
       throw new Error('missing client')
     }
 
-    Logger.debug(TAG + '@handleInitialisePairing', 'pair start')
-    yield* call([client, 'pair'], { uri })
-    Logger.debug(TAG + '@handleInitialisePairing', 'pair end')
+    // Check the client for an existing session with a pairingTopic equal to the uri topic
+    const activeSessions = client.getActiveSessions()
+    const foundSession = Object.keys(activeSessions).find((key) => {
+      return activeSessions[key].pairingTopic === topic
+    })
+
+    if (foundSession) {
+      Logger.debug(TAG + '@handleInitialisePairing', 'extending session start')
+      yield* call([client, 'extendSession'], { topic: foundSession })
```

Comment:
Q: isn't `foundSession` an instance of `Session` (vs `string` which is what I'd expect for `topic`)?